pkg/scheduler/framework/uniquename/uniquename.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// package uniquename features some utilities that are used to generate unique names in use
+// by the scheduler.
+package uniquename
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+const (
+	uuidLength = 6
+)
+
+// minInt returns the smaller one of two integers.
+func minInt(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// ClusterResourceBindingUniqueName returns a unique name for a cluster resource binding in the
+// format of DNS subdomain names (RFC 1123).
+//
+// The name is generated using the following format:
+// * [CRP-NAME] - [TARGET-CLUSTER-NAME] - [RANDOM-SUFFIX]
+//
+// Segments will be truncated if necessary.
+//
+// Note that the name generation is, in essence, a best-effort process, though the chances
+// of name collisions are extremely low.
+//
+// In addition, note that this function assumes that both the CRP name and the cluster name
+// are valid DNS subdomain names (RFC 1123).
+func ClusterResourceBindingUniqueName(CRPName string, clusterName string) (string, error) {
```

Comment:
```suggestion
func NewClusterResourceBindingName(CRPName string, clusterName string) (string, error) {
```
The caller will call this func like "uniquename.ClusterResourceBindingUniqueName" and it has duplicate info there.